runtime/local/src/lib.rs

```diff
@@ -775,17 +775,78 @@ impl pallet_sudo::Config for Runtime {
 )]
 pub enum ProxyType {
     Any,
+    NonTransfer,
+    Governance,
+    CancelProxy,
+    DappsStaking,
 }
 
 impl Default for ProxyType {
     fn default() -> Self {
         Self::Any
     }
 }
+
 impl InstanceFilter<RuntimeCall> for ProxyType {
-    fn filter(&self, _c: &RuntimeCall) -> bool {
+    fn filter(&self, c: &RuntimeCall) -> bool {
         match self {
             ProxyType::Any => true,
+            ProxyType::NonTransfer => {
+                matches!(
+                    c,
+                    RuntimeCall::System(..)
+                        | RuntimeCall::Utility(..)
```

Comment:
Here's hopping Utility doesn't have a bug like we do in our custom sig pallet :)

Reply:
Sorry, could you rephrase it?

Reply:
Our custom sig doesn't check the provided call via filtering.
So if Utility is allowed, and doesn't do filtering, user could just wrap any call into a batch call of size 1 and get it executed.

Reply:
In your senario, `Utility::batch` Call wrap some calls and dispatches them [here](https://github.com/paritytech/substrate/blob/cafbb668d2e57ede83f11f59d8bf291aa8d0c765/frame/utility/src/lib.rs#L223).

When this second dispatch happen, call filtering is triggered again, since what `do_proxy` inside pallet_proxy is doing is to register filtering function to `RuntimeOrigin` itself. [here](https://github.com/paritytech/substrate/blob/cafbb668d2e57ede83f11f59d8bf291aa8d0c765/frame/proxy/src/lib.rs#L774).

This registered filtering function will be applied to all dispatches in the same Tx. https://github.com/paritytech/substrate/blob/cafbb668d2e57ede83f11f59d8bf291aa8d0c765/frame/support/procedural/src/construct_runtime/expand/call.rs#L170

I also confirmed `Utility::batch` extrinsic which contains `Balance::transfer` call inside by `NonTransfer` proxy account will fail due to CallFiltered error.

Reply:
<img width="1503" alt="Screen Shot 2022-11-06 at 15 50 03" src="https://user-images.githubusercontent.com/28953860/200158390-7928b754-3a7a-4c8f-a461-88d1becb058f.png">
ALICE added ALICE_STASH account as NonTransfer proxy.
<img width="1510" alt="Screen Shot 2022-11-06 at 15 54 36" src="https://user-images.githubusercontent.com/28953860/200158407-fd897a38-5725-49cd-af3e-b85f3657504f.png">
Execute batch call with balance transfer by ALICE_STASH fails.